field/generator/internal/templates/extensions/e4.vector.go.tmpl

```diff
@@ -0,0 +1,94 @@
+import (
+	fr "{{ .FieldPackagePath }}"
+	"github.com/consensys/gnark-crypto/utils/cpu"
+)
+
+// Vector represents a vector of E4 elements
+type Vector []E4
+
+func (vector *Vector) Add(a, b Vector) {
+	if len(a) != len(b) || len(a) != len(*vector) {
+		panic("vector.Add: vectors don't have the same length")
+	}
+
+	for i := range a {
+		(*vector)[i].Add(&a[i], &b[i])
+	}
+}
+
+func (vector *Vector) AddMixed(a Vector, b fr.Vector) {
+	if len(a) != len(b) || len(a) != len(*vector) {
+		panic("vector.AddMixed: vectors don't have the same length")
+	}
+	for i := range a {
+		(*vector)[i].Set(&a[i])
+		(*vector)[i].B0.A0.Add(&(*vector)[i].B0.A0, &b[i])
```

Comment:
Does that mean that we hard-cast vectors of base-field elements into vectors of extension field elements?